### DISCUSSION TITLE
Reference property with dot from message template

### DISCUSSION THREAD

**Discussion by MaceWindu** [datalust/seq-tickets#1879] [Help] [answered]
Is it possible with latest Seq version to escape/reformat message template parameters with dot (.) in a way that Seq will be able to substitute them in rendered message?

Trimmed event example. As you can see "obj.x" property cannot be resolved by Seq. I suspect it expects property `obj` with sub-property `x`.

Need to understand wether I can update parameter name generation on my side or I need to change logger call to log in compatible format

```json
{
  "@t":"2023-06-26T18:29:28.5603199Z",
  "@mt":"Test log {@obj.x} or {@y}",
  "@m":"Test log {@obj.x} or 456",
  "obj.x":123,
  "y":456
}
```

**Comment by KodrAus** [comment, marked as answer]
Hi @MaceWindu :wave:

The [message templates spec](https://messagetemplates.org/) doesn't currently support dotted property names, so at this stage to refer to those properties in your message template you'll need to update your client code not to use dots in property names. You can still refer to property names with dots in Seq's query language though. In that case, `obj.x` can be referenced as `@Properties['obj.x']`.

**Reply by MaceWindu** [reply]
Thanks, completely missed that it's against spec. Will fix on my side
